linux/library/include/flutter_desktop_embedding/channels.h

```diff
@@ -0,0 +1,120 @@
+  const std::string &method_name() const { return method_name_; }
+  void set_method_name(const std::string &name) { method_name_ = name; }
+
+  // The arguments to the method call, or a nullValue if there are none.
+  const Json::Value &arguments() const { return arguments_; }
+  void set_arguments(const Json::Value &arguments);
+
+  // Returns a version of the method call serialized in the format expected by
+  // the Flutter engine.
+  Json::Value AsMessage() const;
+
+ private:
+  std::string method_name_;
+  Json::Value arguments_;
+};
+
+// Encapsulates a result sent back to the Flutter engine in response to a
+// MethodCall. Only one method should be called on any given instance.
+class MethodResult {
+ public:
+  // Sends a success response, indicating that the call completed successfully.
+  // An optional value can be provided as part of the success message.
+  void Success(const Json::Value &result = Json::Value());
+
+  // Sends an error response, indicating that the call was understood but
+  // handling failed in some way. A string error code must be provided, and in
+  // addition an optional user-readable error_message and/or details object can
+  // be included.
+  void Error(const std::string &error_code,
+             const std::string &error_message = "",
+             const Json::Value &error_details = Json::Value());
+
+  // Sends a not-implemented response, indicating that the method either was not
+  // recognized, or has not been implemented.
+  void NotImplemented();
+
+ protected:
+  // Internal implementation of the interface methods above, to be implemented
+  // in subclasses.
+  virtual void SuccessInternal(const Json::Value &result) = 0;
+  virtual void ErrorInternal(const std::string &error_code,
+                             const std::string &error_message,
+                             const Json::Value &error_details) = 0;
+  virtual void NotImplementedInternal() = 0;
+};
+
+// Implemention of MethodResult using JSON as the protocol.
+// TODO: Move this logic into method codecs.
+class JsonMethodResult : public MethodResult {
+ public:
+  // Creates a result object that will send results to |engine|, tagged as
+  // associated with |response_handle|. The engine pointer must remain valid for
+  // as long as this object exists.
+  JsonMethodResult(FlutterEngine engine,
+                   const FlutterPlatformMessageResponseHandle *response_handle);
+  ~JsonMethodResult();
+
+ protected:
+  void SuccessInternal(const Json::Value &result) override;
+  void ErrorInternal(const std::string &error_code,
+                     const std::string &error_message,
+                     const Json::Value &error_details) override;
+  void NotImplementedInternal() override;
+
+ private:
+  // Sends the given JSON response object to the engine.
+  void SendResponseJson(const Json::Value &response);
+
+  // Sends the given response data (which must either be nullptr or a serialized
+  // JSON response) to the engine.
```

Comment:
Please comment that this is because it is possible to send a success response that is still an empty string (and that is why this is a pointer instead of a const reference).

Reply:
Done.